From 5ac9c3048611cc8afcfc54c85f5618a562c8f0ec Mon Sep 17 00:00:00 2001
From: Stephane Desneux <stephane.desneux@open.eurogiciel.org>
Date: Mon, 14 Apr 2014 20:44:40 +0200
Subject: [PATCH 13/13] allow to start weston with system user

the check to allow only root user had been replaced
by a check to see if effective uid is equal or less than 499

499 is the actual value in login.defs for SYS_UID_MAX

Change-Id: Ic2b96661df0c939e7f6e066f0bed1d8f7e26bd8f
Signed-off-by: Stephane Desneux <stephane.desneux@open.eurogiciel.org>
---
 src/launcher-util.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/launcher-util.c b/src/launcher-util.c
index 1b63458..47d6eda 100644
--- a/src/launcher-util.c
+++ b/src/launcher-util.c
@@ -160,7 +160,7 @@ weston_launcher_open(struct weston_launcher *launcher,
 	msg.msg_iovlen = 1;
 	msg.msg_control = control;
 	msg.msg_controllen = sizeof control;
-	
+
 	do {
 		len = recvmsg(launcher->fd, &msg, MSG_CMSG_CLOEXEC);
 	} while (len < 0 && errno == EINTR);
@@ -404,7 +404,7 @@ weston_launcher_connect(struct weston_compositor *compositor, int tty,
 					  seat_id, tty);
 		if (r < 0) {
 			launcher->logind = NULL;
-			if (geteuid() == 0) {
+			if (geteuid() <= 499) { /* 499 = SYS_UID_MAX in login.defs, but it should be parsed */
 				if (setup_tty(launcher, tty) == -1) {
 					free(launcher);
 					return NULL;
-- 
1.8.1.4

